cellstore/reader.go

```diff
@@ -176,6 +178,84 @@ func (r *Reader) readBlock(pos int) ([]byte, error) {
 	}
 }
 
+type Near struct {
+	CellID   s2.CellID
+	Value    []byte
+	Distance int32
+}
+
+type Nearby []*Near
+type ByDistance []*Near
+
+func (n ByDistance) Len() int           { return len(n) }
+func (n ByDistance) Swap(i, j int)      { n[i], n[j] = n[j], n[i] }
+func (n ByDistance) Less(i, j int) bool { return n[i].Distance < n[j].Distance }
+
+// FindNearby
+func (r *Reader) FindNearby(loc s2.CellID, limit int) (Nearby, error) {
+	var dst Nearby
+
+	// find the block
+	it, err := r.FindBlock(loc)
+	if err != nil {
+		return dst, err
+	}
+
+	// iterate of the matching block
+	before := 0
+	after := 0
+	for it.Next() {
+		if it.CellID() < loc {
+			before += 1
+		} else {
+			after += 1
+		}
+
+		dist := int32(earthRadius * it.CellID().LatLng().Distance(loc.LatLng()).Radians())
+		dst = append(dst, &Near{CellID: it.CellID(), Value: it.Value(), Distance: dist})
```

Comment:
as said, this doesn't work, you cannot use `Value: it.Value()` as the underlying byte buffer is transient and subject to change with every `Next/NextBlock/PrevBlock` call